ouroboros-consensus/src/Ouroboros/Consensus/Util/Counting.hs

```diff
@@ -66,6 +69,19 @@ import           Ouroboros.Consensus.Util.SOP
 
 newtype Exactly xs a = Exactly { getExactly :: NP (K a) xs }
 
+-- NOTE: This could be a deriving newtype instance, but the constraints will be
+-- a bit more complicated.
+instance forall xs a . (Typeable xs, ToCBOR a, SListI xs) => ToCBOR (Exactly xs a) where
+  toCBOR (Exactly xsTop) = go xsTop
+    where
+      go :: NP (K a) xs' -> Encoding
+      go xs = case xs of
+        Nil        -> mempty
+        K x :* xs' -> toCBOR x <> go xs'
+
+instance (Typeable xs, FromCBOR a, SListI xs) => FromCBOR (Exactly xs a) where
+  fromCBOR = Exactly <$> (htraverse' (\_ -> K <$> fromCBOR) (hpure Proxy))
```

Comment:
Please drop the undue parens. Consider also replacing `_` with `Proxy` or maybe `_Proxy` --- that just makes it slightly clearer what's going on. (I was expecting `core-sop` to provide a specialized function for this, but it's not so surprising it's not there.)

Reply:
Just FYI, I don't have a preference between `Proxy` and `K ()`. I personally think of both as https://hackage.haskell.org/package/base-4.14.1.0/docs/GHC-Generics.html#t:U1 FWIW, but that's probably a b/c of my personal chronology.